src/nlp/arrow_dataset.py

```diff
@@ -557,3 +558,39 @@ def apply_function_on_filtered_inputs(inputs, indices):
                 return Dataset.from_buffer(buf_writer.getvalue())
         else:
             return self
+
+    def filter(self, function, with_indices=False, **kwargs):
+        def map_function(batch, *args):
+            result = defaultdict(list)
+            num_examples = len(batch[next(iter(batch.keys()))])
+
+            # create single examples
+            for i in range(num_examples):
+                example = map_nested(lambda x: x[i], batch, dict_only=True)
```

Comment:
So we assume that `batch` is a `dict{key_1: list, key_2: list, ...}`. With `dict_only=True` this means that the function has to be applied directly to every list in the `dict` and it should be fine @lhoestq

Reply:
Ok I see what you mean !